libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -262,6 +262,16 @@ class MapboxNavigation(
         tripSession.stop()
     }
 
+    /**
+     * Reset the session with the same configuration. The location becomes unknown,
+     * but the [NavigationOptions] stay the same. This can be used to transport the
+     * navigator to a new location.
+     */
+    fun resetTripSession() {
+        directionsSession.routes = emptyList()
```

Comment:
Do we need to reset the route as well? I'd expect this method to only reset the Navigator's bias (we can work on the method naming as well) and nothing else.

Reply:
While reviewing this noticed that we were not resetting the `directionsSession.routes` as part of `onDestroy` which allows you to do

```kotlin
mapboxNavigation.onDestroy()
mapboxNavigation.getRoutes()
```

We considered adding it here which is called when in `onDestroy`. If we opt to remove it we should add it after `directionsSession.unregisterAllRoutesObservers()` when in `onDestroy` (we can do it in a separate PR though as it doesn't directly relate to the changes added as part of this PR).

cc @kmadsen

Reply:
We agreed in comments that native was going to reset the route, and the SDK callers would need to set route again if they wanted to keep it. But that is actually, not true.

``` c++
void NavigatorImpl::resetRideSession() {
  BuildRideSessionImpl();
  if (routes_ && ride_session_) {
    try {
      ride_session_->SetRoutes(routes_);
    } catch (const std::exception& ex) {
      LOG_ERROR(std::string{"SetRoute error: "} + ex.what());
    }
  }
```

cc: @mskurydin @SiarheiFedartsou @averkhaturau

I think @LukasPaczos is right, we don't want to reset the route inside this `resetTripSession` function. So am moving the route reset back to `onDestroy`